app/uk/gov/hmrc/apidefinition/services/APIDefinitionService.scala

```diff
@@ -62,6 +67,27 @@ class APIDefinitionService @Inject()(wso2Publisher: WSO2APIPublisher,
     }
   }
 
+  private def checkAPIDefinitionForStatusChanges(apiDefinition: APIDefinition): Unit = {
+    def findStatusDifferences(existingAPIVersions: Seq[APIVersion], newAPIVersions: Seq[APIVersion]): Future[Seq[(String, APIStatus, APIStatus)]] =
+      Future {
+        (existingAPIVersions ++ newAPIVersions)
+          .groupBy(_.version)
+          .filter(v => v._2.size == 2)
+          .filterNot(v => v._2.head.status == v._2.last.status)
+          .map(v => (v._1, v._2.head.status, v._2.last.status))
+          .toSeq
+      }
+
+    for {
+      apiDefinitionOption <- apiDefinitionRepository.fetchByName(apiDefinition.name)
```

Comment:
I think the name can change. It may be better to fetch by context as we don't allow to change the context.